### DISCUSSION TITLE
[@types/node] readline/promises not found

### DISCUSSION THREAD

**Discussion by Didas-git** [DefinitelyTyped/DefinitelyTyped#59382] [Issues with a @types package]
I was trying to test the promises api for readline but the types cant be found even tho the types file says its possible to.

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
I think it's not implemented, you're probably referring to existing JSDoc comments about using `readline/promises`, but AFAIK readline module awaits contribution for this

**Reply by Didas-git** [reply]
> I think it's not implemented, you're probably referring to existing JSDoc comments about using `readonly/promises`, but AFAIK readline module awaits contribution for this

Acordding to the nodejs docs it was implemented in v17
Reference: https://nodejs.org/api/readline.html#promises-api

**Reply by peterblazejewicz** [reply]
Yes, that's correct. My point is there is no TS implementation here for @types/node for this specific module yet.

**Reply by Didas-git** [reply]
I thought so, but i decided to create this discussion specially because on the realine.d.ts it says you can import the promises version of it which is kind of misleading
